server/src/main/java/io/crate/rest/action/SqlHttpHandler.java

```diff
@@ -294,9 +296,26 @@ private CompletableFuture<XContentBuilder> executeBulkRequest(Session session,
             });
     }
 
+    /**
+     * Doesn't do authentication.
+     * Checks user existence and if not possible to resolve from header (basic or jwt),
+     * returns trusted user from configuration.
+     */
     Role userFromAuthHeader(@Nullable String authHeaderValue) {
-        try (Credentials credentials = Headers.extractCredentialsFromHttpBasicAuthHeader(authHeaderValue)) {
-            String username = credentials.username();
+        try (Credentials credentials = Headers.extractCredentialsFromHttpAuthHeader(authHeaderValue)) {
+            String username = null;
+            var decodedJwt = credentials.decodedToken();
+            if (decodedJwt != null) {
+                Role role = roles.findUser(
+                    decodedJwt.getIssuer(),
+                    decodedJwt.getClaim("username").asString()
+                );
+                if (role != null) {
+                    username = role.name();
+                }
+            } else {
+                username = credentials.username();
+            }
```

Comment:
Shouldn't this also verify the token?

Reply:
No, because it's same as password - otherwise we need to reason why we don't check password for _sql endpoint.

I think what happens for _sql endpoint is described in 
https://cratedb.com/docs/crate/reference/en/5.6/admin/auth/methods.html#trust-authentication-over-http
I BTW need to update the docs 
The HTTP implementation extracts the username from the HTTP Basic Authentication -> should be Basic or Bearer

Reply:
Also, do you think we need to explicitly document that _sql endpoint does trust over http in https://cratedb.com/docs/crate/reference/en/5.6/interfaces/http.html?

Reply:
Actually, at this point authentication is done, as  `HttpAuthUpstreamHandler` is registered before the `SqlHttpHandler`, so it's fine do align with Basic header handling - only resolve username from header.